### ISSUE TITLE
ERROR:    Exception in ASGI application

### ISSUE THREAD

**Issue by YashaswiniIppili** [gradio-app/gradio#5969]
### Describe the bug

So I'm running this LLM on colab that's a chatbot based on Gen AI, transformers, huggingface, etc. And the interface isn't working, I have no clue on how to fix it. It was working fine perfectly 2 weeks ago.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

So I'm running this LLM on colab that's a chatbot based on Gen AI, transformers, huggingface, etc. And the interface isn't working, I have no clue on how to fix it. It was working fine perfectly 2 weeks ago.

### Screenshot

![image](https://github.com/gradio-app/gradio/assets/92773216/82143657-18c4-43e4-a2d1-7af4608903c6)

### Logs

```shell
Colab notebook detected. This cell will run indefinitely so that you can see errors and logs. To turn off, set debug=False in launch().
Running on public URL: https://200ade20f32d66898b.gradio.live

This share link expires in 72 hours. For free permanent hosting and GPU upgrades, run `gradio deploy` from Terminal to deploy to Spaces (https://huggingface.co/spaces)
Streaming output truncated to the last 5000 lines.
    raise InvalidURL("No scheme included in URL.")
httpx._exceptions.InvalidURL: No scheme included in URL.
ERROR:    Exception in ASGI application
```

### System Info

```shell
Yes, I'm using the latest gradio version.
```

### Severity

Blocking usage of gradio

**Comment by abidlabs** [comment]
Hi @YashaswiniIppili can you share the code for a reproducible example?

**Comment by YashaswiniIppili** [comment]
import gradio as gr
import torch
from llama_index import VectorStoreIndex, SimpleDirectoryReader, ServiceContext, LangchainEmbedding
from llama_index.llms import HuggingFaceLLM
from llama_index.prompts.prompts import SimpleInputPrompt
from llama_index.readers import SimpleDirectoryReader
from langchain.embeddings.huggingface import HuggingFaceEmbeddings
from googletrans import Translator
import re
import os
import time

# Initialize LLM and other models
llm = HuggingFaceLLM(
    context_window=4096,
    max_new_tokens=256,
    generate_kwargs={"temperature": 0.0, "do_sample": False},
    system_prompt="You are a Q&A assistant. Your goal is to answer questions as accurately as possible based on the instructions and context provided.",
    query_wrapper_prompt=SimpleInputPrompt("{query_str}"),
    tokenizer_name="meta-llama/Llama-2-7b-chat-hf",
    model_name="meta-llama/Llama-2-7b-chat-hf",
    device_map="auto",
    model_kwargs={"torch_dtype": torch.float16, "load_in_8bit": True}
)

operations_model = LangchainEmbedding(
    HuggingFaceEmbeddings(model_name="sentence-transformers/all-mpnet-base-v2")
)

installation_model = LangchainEmbedding(
    HuggingFaceEmbeddings(model_name="sentence-transformers/all-mpnet-base-v2")
)

# manual_model = LangchainEmbedding(
#     HuggingFaceEmbeddings(model_name="sentence-transformers/all-mpnet-base-v2")
# )

# service_model = LangchainEmbedding(
#     HuggingFaceEmbeddings(model_name="sentence-transformers/all-mpnet-base-v2")
# )

# Define the document folders for operations and installation
operations_documents_folder = "/content/drive/MyDrive/GE/operation"
installation_documents_folder = "/content/drive/MyDrive/GE/installation"
# manual_documents_folder = "/content/drive/MyDrive/GE/manuals"
# service_documents_folder = "/content/drive/MyDrive/GE/service"

# Create a dictionary to map user input to the corresponding model and document folder
model_mapping = {
    "Operations": (operations_model, operations_documents_folder),
    "Installation": (installation_model, installation_documents_folder),
    # "Manuals": (manual_model, installation_documents_folder),
    # "Service": (service_model, installation_documents_folder)
}

# Initialize the translator
translator = Translator()

# Define a function to save the query and response to the transcript file
def save_to_transcript(query, response):
    with open("transcript.txt", "a", encoding="utf-8") as file:
        file.write(f"User: {query}\nAssistant: {response}\n\n")

# Define the Gradio interface for the chat component
def chat_interface(model_selection, query, search):
    if query == "":
        # This is the initial greeting message
        return "Hello, How may I assist you today?"
    if search:
        selected_model, documents_folder = model_mapping[model_selection]

        # Load documents from the selected folder
        documents = SimpleDirectoryReader(documents_folder).load_data()

        # Set up the service context with the selected model
        service_context = ServiceContext.from_defaults(
            chunk_size=1024,
            llm=llm,
            embed_model=selected_model,
        )
        index = VectorStoreIndex.from_documents(documents, service_context=service_context)
        query_engine = index.as_query_engine()

        # Detect the language of the input query
        input_lang = translator.detect(query).lang

        # Translate the query to English if it's not already in English
        if input_lang != 'en':
            query = translator.translate(query, src=input_lang, dest='en').text

        # Send the translated query to your query engine
        response = query_engine.query(query)

        # Translate the response back to the original language
        response = translator.translate(response, src='en', dest=input_lang).text

        # Replace URLs with clickable links
        response = re.sub(r'(https?://\S+)', r'<a href="\1">\1</a>', response)

        # Check if the user's response contains "thank you" or "bye" and respond accordingly
        if "thank you" in query.lower():
            response += "\nI'm happy to help. Please go to https://www.gehealthcare.com/support/manuals for further reference."
        elif "bye" in query.lower():
            response += "\nGoodbye! If you have more questions in the future, feel free to ask."
        elif any(greeting in query.lower() for greeting in ["hey", "hi", "hello"]):
            response = "Hello! How may I assist you today?"

        # Save the query and response to the transcript file
        save_to_transcript(query, response)

        # Return the response
        return response
    else:
        # This is the initial greeting message
        return "Hello, How may I help you today?"

# Define the Gradio chatbot interface
def respond(message, chat_history, selected_option):
    bot_message = chat_interface(selected_option, message, True)
    chat_history.append((message, bot_message))
    time.sleep(2)  # Simulate some processing time
    return "", chat_history

# Create the Gradio interface using Blocks
with gr.Blocks() as demo:
    gr.Markdown('GEHC Chatbot')
    options = ["Operations", "Installations", "Manuals", "Service"]
    selected_option = gr.Dropdown(options, label="Select Category")

    # Add the chatbot and textbox components
    chatbot = gr.Chatbot()
    msg = gr.Textbox()
    clear = gr.ClearButton([msg, chatbot])
    # chatbot.value = "Hello! How may I assist you today?"

    # Set up the submit button to call the respond function with the selected option
    msg.submit(respond, [msg, chatbot, selected_option], [msg, chatbot])

demo.launch(share=True, debug=True)

**Comment by YashaswiniIppili** [comment]
> Hi @YashaswiniIppili can you share the code for a reproducible example?

Can you please check this out and lemme know what the issue is ? 🙏🏻

**Comment by YashaswiniIppili** [comment]
> Hi @YashaswiniIppili can you share the code for a reproducible example?

# Define the Gradio interface for the chat component
def chat_interface(model_selection, query, search):
    if query == "":
        # This is the initial greeting message
        return "Hello, How may I assist you today?"
    if search:
        selected_model, documents_folder = model_mapping[model_selection]

        # Load documents from the selected folder
        documents = SimpleDirectoryReader(documents_folder).load_data()

        # Set up the service context with the selected model
        service_context = ServiceContext.from_defaults(
            chunk_size=1024,
            llm=llm,
            embed_model=selected_model,
        )
        index = VectorStoreIndex.from_documents(documents, service_context=service_context)
        query_engine = index.as_query_engine()

        # Detect the language of the input query
        input_lang = translator.detect(query).lang

        # Translate the query to English if it's not already in English
        if input_lang != 'en':
            query = translator.translate(query, src=input_lang, dest='en').text

        # Send the translated query to your query engine
        response = query_engine.query(query)

        # Translate the response back to the original language
        response = translator.translate(response, src='en', dest=input_lang).text

        # Replace URLs with clickable links
        response = re.sub(r'(https?://\S+)', r'<a href="\1">\1</a>', response)

        # Check if the user's response contains "thank you" or "bye" and respond accordingly
        if "thank you" in query.lower():
            response += "\nI'm happy to help. Please go to https://www.gehealthcare.com/support/manuals for further reference."
        elif "bye" in query.lower():
            response += "\nGoodbye! If you have more questions in the future, feel free to ask."
        elif any(greeting in query.lower() for greeting in ["hey", "hi", "hello"]):
            response = "Hello! How may I assist you today?"

        # Save the query and response to the transcript file
        save_to_transcript(query, response)

        # Return the response
        return response
    else:
        # This is the initial greeting message
        return "Hello, How may I help you today?"

# Define the Gradio chatbot interface
def respond(message, chat_history, selected_option):
    bot_message = chat_interface(selected_option, message, True)
    chat_history.append((message, bot_message))
    time.sleep(2)  # Simulate some processing time
    return "", chat_history

# Create the Gradio interface using Blocks
with gr.Blocks() as demo:
    gr.Markdown('GEHC Chatbot')
    options = ["Operations", "Installations", "Manuals", "Service"]
    selected_option = gr.Dropdown(options, label="Select Category")

    # Add the chatbot and textbox components
    chatbot = gr.Chatbot()
    msg = gr.Textbox()
    clear = gr.ClearButton([msg, chatbot])
    # chatbot.value = "Hello! How may I assist you today?"

    # Set up the submit button to call the respond function with the selected option
    msg.submit(respond, [msg, chatbot, selected_option], [msg, chatbot])

demo.launch(share=True, debug=True)

**Comment by neeerp** [comment]
> > Hi @YashaswiniIppili can you share the code for a reproducible example?
> 
> Can you please check this out and lemme know what the issue is ? 🙏🏻

Hey @YashaswiniIppili, the example you provided doesn't reproduce the issue... Namely, you haven't provided the definition for `model_mapping`.

**Comment by TuTuHuss** [comment]
@YashaswiniIppili  I encountered the same problem, the reason is that I used `googletrans.Translate()` in my code, which depends on `httpx==0.13.3`, and gradio uses `0.25.0`. Maybe there are other installations or libraries that cause the httpx version to be other than 0.25.0, you can check the httpx version in your python environment, and try `pip install httpx==0.25.0`

**Comment by abidlabs** [comment]
Closing for lack of repro. @YashaswiniIppili if you share the code you are running, we may be able to help.

**Comment by YashaswiniIppili** [comment]
Thank you @TuTuHuss !!

**Comment by daviibf** [comment]
Precisely @TuTuHuss ! Thanks !!!
